src/odemis/gui/comp/viewport.py

```diff
@@ -854,84 +860,124 @@ def setView(self, view, tab_data):
 
         self._view = view
         self._tab_data_model = tab_data
-        # current stream displayed (or None)
-        self.stream = None
 
         # canvas handles also directly some of the view properties
         self.canvas.setView(view, tab_data)
 
         view.stream_tree.flat.subscribe(self.connect_stream)
 
-        self.bottom_legend.set_pol_callback(self.on_legend_pol_change)
-
-    def connect_stream(self, streams):
-        """ Find the most appropriate stream in the view to be displayed, and make sure the display
-        is updated when the stream is updated.
+        self.bottom_legend.set_callback(self.on_legend_change)
 
+    def connect_stream(self, projs):
         """
-        if not streams:
-            stream = None
-        elif len(streams) > 1:
-            logging.warning("Found %d streams, will pick one randomly", len(streams))
-            if self.stream in streams:
-                stream = self.stream  # don't change
-            else:
-                stream = streams[0]
+        Called when the stream_tree is changed.
+        :param projs: (list of Streams or Projections)
+        Find the most appropriate stream in the view to be displayed, and make
+        sure the display is updated when the stream is updated.
+        """
+        if not projs:
+            proj = None
+        elif len(projs) > 1:
+            # => pick the first one playing
+            for o in projs:
+                s = get_original_stream(o)
+                if s.should_update.value:
+                    proj = o
+                    break
+            else:  # no stream playing
+                logging.warning("Found %d streams, will pick one randomly", len(projs))
+                if self._projection in projs:
+                    proj = self._projection  # don't change
+                else:
+                    proj = projs[0]
         else:
-            stream = streams[0]
+            proj = projs[0]
 
-        if self.stream is stream:
-            logging.debug("Not reconnecting to stream as it's already connected")
+        if self._projection is proj:
+            # logging.debug("not reconnecting to stream as it's already connected")
```

Comment:
I'd suggest to leave this logging, as it might help to understand some bugs if the wrong projection is shown in the GUI.

Reply:
Mhm not sure how this happened. Did not work on this code. Guess happened accidentally when searching for something.